packages/rhf-mui/src/utils.ts

```diff
@@ -0,0 +1,60 @@
+type UnionToIntersection<U> = (U extends any ? (k: U) => void : never) extends (
+  k: infer I
+) => void
+  ? I
+  : never
+
+function pushProp(target: {[key: string]: any}, key: string, value: any): void {
+  if (key === 'className') {
+    target.className = [target.className, value].join(' ').trim()
+  } else if (key === 'style') {
+    target.style = {...target.style, ...value}
+  } else if (typeof value === 'function') {
+    const oldFn = target[key] as Function | undefined
+    target[key] = oldFn
+      ? (...args: any[]) => {
+          oldFn(...args)
+          ;(value as Function)(...args)
+        }
+      : value
+  } else if (
+    // skip merging undefined values
+    value === undefined ||
+    // skip if both value are the same primitive value
+    (typeof value !== 'object' && value === target[key])
+  ) {
+    return
+  } else if (!(key in target)) {
+    target[key] = value
+  } else {
+    mergeProps(target[key], value)
```

Comment:
Inline copy of [`merge-props`](https://github.com/andrewbranch/merge-props/blob/master/src/index.ts) to allow deep nested props